src/client.ts

```diff
@@ -1315,9 +1314,14 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
     if (!this.wsConnection && (this.options.warmUp || this.options.enableInsights)) {
       this._sayHi();
     }
-
     // The StableWSConnection handles all the reconnection logic.
-    this.wsConnection = new StableWSConnection<StreamChatGenerics>({ client: this });
+    if (this.options.wsConnection && this.node) {
+      // Intentionally avoiding adding ts generics on wsConnection in options since its only useful for unit test purpose.
+      ((this.options.wsConnection as unknown) as StableWSConnection<StreamChatGenerics>).setClient(this);
+      this.wsConnection = (this.options.wsConnection as unknown) as StableWSConnection<StreamChatGenerics>;
```

Comment:
This is purely to ease unit-testing around connection logic. Currently you can't really properly mock connection class methods since we always create a new instance of `StableWSConnection` class.
Allowing `wsConnection` in options will help with writing more unit tests.